### PR TITLE
Put validators on requests

### PR DESCRIPTION
**Pull request by onats1**

### CODE REVIEW

File: pos-client-library/src/main/java/com/globalaccelerex/globalaccelerexandroidposclientlibrary/transactions/CardNotPresentTransactions.kt [hunk 1]
```diff
@@ -31,6 +31,8 @@ class CardNotPresentTransactions(countryCode: Countries) {
         customPrint: Boolean,
         callingComponent: Any
     ) {
+        require(cardNumber.length in 12..18) {throw IllegalArgumentException("Card Number should be between 12 and 18 digits")}
```
Comment (odifek) [inline pos-client-library/src/main/java/com/globalaccelerex/globalaccelerexandroidposclientlibrary/transactions/CardNotPresentTransactions.kt:34]: require already throws an IllegalArgumentException. No need to throw another one.

**Review by odifek — COMMENTED** [review]
Please check.

File: pos-client-library/src/main/java/com/globalaccelerex/globalaccelerexandroidposclientlibrary/transactions/CardNotPresentTransactions.kt (hunk 1 above)
Comment (onats1) [reply pos-client-library/src/main/java/com/globalaccelerex/globalaccelerexandroidposclientlibrary/transactions/CardNotPresentTransactions.kt:34]: This is to pass the information that caused the crash

Comment (odifek) [reply pos-client-library/src/main/java/com/globalaccelerex/globalaccelerexandroidposclientlibrary/transactions/CardNotPresentTransactions.kt:34]: Yes, just write the message inside the lambda. Not need to throw any other exception. If you check the signature of require, it uses the string in the lambda to throw the exception

Comment (onats1) [reply pos-client-library/src/main/java/com/globalaccelerex/globalaccelerexandroidposclientlibrary/transactions/CardNotPresentTransactions.kt]: I've updated it.

**Review by odifek — APPROVED** [review]

**Pull request merged**